frontend/javascripts/oxalis/controller/combinations/tool_controls.ts

```diff
@@ -780,6 +784,171 @@ export class QuickSelectTool {
   static onToolDeselected() {}
 }
 
+export class LineMeasurementTool {
+  static DOUBLE_CLICK_TIME_THRESHOLD = 600;
+  static getPlaneMouseControls(_planeId: OrthoView): any {
+    let initialPlane: OrthoView = OrthoViews.PLANE_XY;
+    let isMeasuring = false;
+    let lastLeftClickTime = 0;
+    const SceneController = getSceneController();
+    const { lineMeasurementGeometry } = SceneController;
+    const mouseMove = (
+      _delta: Point2,
+      pos: Point2,
+      plane: OrthoView | null | undefined,
+      evt: MouseEvent,
+    ) => {
+      // In case the tool was reset by the user, abort measuring.
+      if (lineMeasurementGeometry.wasReset && isMeasuring) {
+        isMeasuring = false;
+      }
+      if (plane !== initialPlane || !isMeasuring) {
+        MoveHandlers.moveWhenAltIsPressed(_delta, pos, plane, evt);
+        return;
+      }
+      const state = Store.getState();
+      const newPos = V3.floor(calculateGlobalPos(state, pos, initialPlane));
+      lineMeasurementGeometry.setTopPoint(newPos);
+      Store.dispatch(showMeasurementTooltipAction([evt.clientX, evt.clientY]));
+    };
+    const rightClick = (pos: Point2, plane: OrthoView, event: MouseEvent) => {
+      // In case the tool was reset by the user, abort measuring.
+      if (lineMeasurementGeometry.wasReset && isMeasuring) {
+        isMeasuring = false;
+        return;
+      }
+      if (isMeasuring) {
+        // Set the last point of the measurement and stop measuring.
+        mouseMove({ x: 0, y: 0 }, pos, plane, event);
+        isMeasuring = false;
+      } else {
+        // If the tool already stopped measuring, reset the tool.
+        lineMeasurementGeometry.reset();
+        lineMeasurementGeometry.hide();
+        Store.dispatch(hideMeasurementTooltipAction());
+      }
+    };
+    const leftClick = (pos: Point2, plane: OrthoView, event: MouseEvent) => {
+      // In case the tool was reset by the user, abort measuring.
+      if (lineMeasurementGeometry.wasReset && isMeasuring) {
+        isMeasuring = false;
+        return;
+      }
+      const currentTime = Date.now();
+      if (currentTime - lastLeftClickTime <= this.DOUBLE_CLICK_TIME_THRESHOLD) {
```

Comment:
wouldn't it make sense to check that the two clicks were registered roughly at the same spot?